src/otp-input/otp-input.tsx

```diff
@@ -190,19 +193,33 @@ export const OtpInput = ({
             : `${name}-${index + 1}`;
     };
 
-    // =============================================================================
-    // RENDER FUNCTIONS
-    // =============================================================================
-    const renderCTALabel = () => {
+    const displaySeconds = (time: number) => {
+        return `${time} second${time === 1 ? "" : "s"}`;
+    };
+
+    const getCTALabelProps = (): Partial<
+        React.HTMLAttributes<HTMLButtonElement>
+    > => {
         return otherCtaProps.children
-            ? otherCtaProps.children
-            : `Resend OTP${
-                  countDown
-                      ? ` in ${countDown} second${countDown > 1 ? "s" : ""}`
-                      : ""
-              }`;
+            ? { children: otherCtaProps.children }
+            : {
+                  children:
+                      countDown > 0
+                          ? `Resend OTP in ${displaySeconds(countDown)}`
+                          : "Resend OTP",
+                  // use a fixed label to avoid repeated screen reader updates when the button is focused
+                  "aria-label":
+                      countDown > 0
+                          ? `Resend OTP in ${displaySeconds(cooldownDuration)}`
+                          : "Resend OTP",
+                  // use a secondary element to reflect the latest countdown
+                  "aria-describedby": timerId,
```

Comment:
works well in safari but chrome skips it, though at the end of the countdown, it still announces `Ready to resend OTP`

Reply:
thanks so much for catching this! after some further tests with Voiceover:
1. `aria-describedby` doesn't get read on the button when the describing element has a `role="timer"` set in Safari
2. `aria-description` works in the latest Safari but it's still fairly new, so `aria-describedby` is still recommended
3. Chrome was skipping the button depending on where the automatic read-all function started. If you start on an element before the OtpInput, it works fine; if you start within the OtpInput, it skips the button. I found that moving the visually hidden items to render before the button fixes that somehow 🙃

Reply:
wow thanks for doing a deeper dive into this, I also learnt new things today :)